beacon-chain/powchain/service.go

```diff
@@ -505,19 +505,31 @@ func (s *Service) initDepositCaches(ctx context.Context, ctrs []*protodb.Deposit
 		return nil
 	}
 	s.depositCache.InsertDepositContainers(ctx, ctrs)
-	currentState, err := s.beaconDB.HeadState(ctx)
-	if err != nil {
-		return errors.Wrap(err, "could not get head state")
-	}
-	// do not add to pending cache
-	// if no state exists.
-	if currentState == nil {
+	if !s.chainStartData.Chainstarted {
+		// do not add to pending cache
+		// if no genesis state exists.
 		validDepositsCount.Add(float64(s.preGenesisState.Eth1DepositIndex() + 1))
 		return nil
 	}
-	currIndex := currentState.Eth1DepositIndex()
+	genesisState, err := s.beaconDB.GenesisState(ctx)
+	if err != nil {
+		return err
+	}
+	// Default to all deposits post-genesis deposits in
+	// the event we cannot find a suitable head state.
+	currIndex := genesisState.Eth1DepositIndex()
+	rt := s.beaconDB.LastArchivedRoot(ctx)
+	if rt != [32]byte{} {
+		currentState, err := s.beaconDB.State(ctx, rt)
+		if err != nil {
+			return errors.Wrap(err, "could not get head state")
```

Comment:
nitpick, this is last archived state

```suggestion
			return errors.Wrap(err, "could not get last archived state")
```